templates/rhdm70-full-persistent.yaml

```diff
@@ -0,0 +1,508 @@
+- kind: Route
+  apiVersion: v1
+  id: "${APPLICATION_NAME}-rhdmcentr-http"
+  metadata:
+    name: "${APPLICATION_NAME}-rhdmcentr"
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: Route for Decision Central's http service.
+  spec:
+    host: "${DECISION_CENTRAL_HOSTNAME_HTTP}"
+    to:
+      name: "${APPLICATION_NAME}-rhdmcentr"
+- kind: Route
+  apiVersion: v1
+  id: "${APPLICATION_NAME}-rhdmcentr-https"
+  metadata:
+    name: secure-${APPLICATION_NAME}-rhdmcentr
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: Route for Decision Central's https service.
+  spec:
+    host: "${DECISION_CENTRAL_HOSTNAME_HTTPS}"
+    to:
+      name: secure-${APPLICATION_NAME}-rhdmcentr
+    tls:
+      termination: passthrough
+- kind: Route
+  apiVersion: v1
+  id: "${APPLICATION_NAME}-kieserver-http"
+  metadata:
+    name: "${APPLICATION_NAME}-kieserver"
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: Route for KIE server's http service.
+  spec:
+    host: "${EXECUTION_SERVER_HOSTNAME_HTTP}"
+    to:
+      name: "${APPLICATION_NAME}-kieserver"
+- kind: Route
+  apiVersion: v1
+  id: "${APPLICATION_NAME}-kieserver-https"
+  metadata:
+    name: secure-${APPLICATION_NAME}-kieserver
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: Route for KIE server's https service.
+  spec:
+    host: "${EXECUTION_SERVER_HOSTNAME_HTTPS}"
+    to:
+      name: secure-${APPLICATION_NAME}-kieserver
+    tls:
+      termination: passthrough
+- kind: DeploymentConfig
+  apiVersion: v1
+  metadata:
+    name: "${APPLICATION_NAME}-rhdmcentr"
+    labels:
+      application: "${APPLICATION_NAME}"
+  spec:
+    strategy:
+      type: Recreate
+    triggers:
+    - type: ImageChange
+      imageChangeParams:
+        automatic: true
+        containerNames:
+        - "${APPLICATION_NAME}-rhdmcentr"
+        from:
+          kind: ImageStreamTag
+          namespace: "${IMAGE_STREAM_NAMESPACE}"
+          name: rhdm70-decisioncentral-openshift:dev
+    - type: ConfigChange
+    replicas: 1
+    selector:
+      deploymentConfig: "${APPLICATION_NAME}-rhdmcentr"
+    template:
+      metadata:
+        name: "${APPLICATION_NAME}-rhdmcentr"
+        labels:
+          deploymentConfig: "${APPLICATION_NAME}-rhdmcentr"
+          application: "${APPLICATION_NAME}"
+      spec:
+        serviceAccountName: rhdm-service-account
+        terminationGracePeriodSeconds: 60
+        containers:
+        - name: "${APPLICATION_NAME}-rhdmcentr"
+          image: rhdm70-decisioncentral-openshift
+          imagePullPolicy: Always
+          volumeMounts:
+          - name: rhdm-keystore-volume
+            mountPath: "/etc/rhdm-secret-volume"
+            readOnly: true
+          - name: "${APPLICATION_NAME}-rhdmcentr-pvol"
+            mountPath: "/opt/eap/standalone/data/bpmsuite"
```

Comment:
I think that we should also add a new volume for application users and roles. In /opt/eap/standalone/configuration are stored properties files with users and roles. Now, if we restart DM deployment, all configuration is lost.

Reply:
@jakubschwan AFAIK users and roles are recreated every time the container is started, so there is no point in storing them.

Reply:
AFAICT it will create two pods - one for the kie server and one for the decision central.